qiskit_machine_learning/neural_networks/opflow_qnn.py

```diff
@@ -114,6 +114,24 @@ def _construct_gradient_operator(self):
         except (ValueError, TypeError, OpflowError, QiskitError):
             logger.warning("Cannot compute gradient operator! Continuing without gradients!")
 
+    @property
+    def quantum_instance(self) -> QuantumInstance:
+        """Returns the quantum instance to evaluate the circuit."""
+        return self._quantum_instance
+
+    @quantum_instance.setter
```

Comment:
I think the same applies here where this setter and code in the constructor are duplicated. Or almost. In the setter here if its not passed a QuantumInstance, but rather a [Base]Backend it does not make a QuantumInstance out of it, like I think it should, as such the getter will return a [Base]Backend which is not what we want I think. I note that CircuitSampler, like we normally do, allows a QuantimInstance or {Base]Backend to be passed in so the code works in that regard, but it should be wrapped to a QuantumInstance if its not already, like the equivalent code in the constructor does.

Arguably there is a bug in the constructor code that the quantum_instance test should be explicit to be is not None, like is done here.

Reply:
In case the 'I think the same applies here....' is not clear I had made it while appending a comment to one or the review bits above on circuit_qnn here https://github.com/Qiskit/qiskit-machine-learning/pull/79#discussion_r644339176  I see these comments are far apart now; at least on the page - but they were done at the same time as you can see from the timestamps.

Reply:
I noticed that we were assigning `_sampler` in the constructor, but `_circuit_sampler` in the setter, which I think was an error. In the refactored version we assign just `self._circuit_sampler` as a `CircuitSampler` instance, let me know if that's incorrect. 

Also I assume we wish to similarly wrap a `Backend` in a `QuantumInstance` within the setter of `circuit_qnn.py` as well, so I made the change there as well